src/main/java/dude/Ui.java

```diff
@@ -0,0 +1,156 @@
+package dude;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import dude.exception.DudeException;
+import dude.task.Task;
+import dude.task.TaskList;
+
+/**
+ * Represents the user interface for interacting with the user.
+ * This class handles input from the user and displays messages.
+ */
+public class Ui {
+    private static final String BOT_NAME = "Dude";
+    private static final String LINE = "____________________________________________________________";
+    private Scanner scanner;
+
+    /**
+     * Constructs a Ui and initializes the Scanner for user input.
+     */
+    public Ui() {
+        scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Displays the greeting message.
+     */
+    public void showGreet() {
+        System.out.println(LINE);
+        System.out.println("Hello! I'm " + BOT_NAME);
+        System.out.println("What can I do for you?");
+        System.out.println(LINE);
+    }
```

Comment:
I really like how your methods follow the coding standard. i.e method name is a verb